li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/liazkaban/LiAzkabanDslCompiler.groovy

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.gradle.liazkaban;
+
+
+import com.linkedin.gradle.azkaban.AzkabanDslCompiler;
+import com.linkedin.gradle.hadoopdsl.job.Job;
+import com.linkedin.gradle.libangbang.LiHadoopShellCommand;
+import com.linkedin.gradle.libangbang.LiHadoopShellCommandFactory;
+import com.linkedin.gradle.lihadoopdsl.lijob.LiBangBangJob;
+import com.linkedin.gradle.lihadoopdsl.lijob.LiPigBangBangJob;
+import groovy.text.GStringTemplateEngine;
+import groovy.text.Template;
+import org.gradle.api.Project;
+
+class LiAzkabanDslCompiler extends AzkabanDslCompiler {
+
+  private static final String BANGBANG_TEMPLATE = "LiBangBangJob.template";
+  private LiHadoopShellCommandFactory bangBangCommandFactory;
+
+  /**
+   * Constructor for the AzkabanDslCompiler.
+   *
+   * @param project The Gradle project
+   */
+  LiAzkabanDslCompiler(Project project) {
+    super(project);
+  }
+
+  /**
+   * Separate visitor for the BangBang type pig job to build and write gradle file
+   * @param job The LiPigBangBangJob to build
+   */
+  void visitJobToBuild(LiPigBangBangJob job) {
+    writeGradleForJob(job);
+    writeJobFile(job);
+  }
+
+  /**
+   * This file duplicates functionality from the superclass. We have to remove other
+   * properties from the job file apart from the command and type
+   * @param job The bangbang job
+   */
+  void writeJobFile(job) {
```

Comment:
Can you add the type to the job argument? Although you don't need it in Groovy, I like all the functions to be properly typed